pkgs/by-name/we/wemeet/package.nix

```diff
@@ -0,0 +1,253 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  dpkg,
+  autoPatchelfHook,
+  makeWrapper,
+  nss,
+  xorg,
+  desktop-file-utils,
+  libpulseaudio,
+  libgcrypt,
+  dbus,
+  systemd,
+  udev,
+  libGL,
+  fontconfig,
+  freetype,
+  openssl,
+  wayland,
+  libdrm,
+  harfbuzz,
+  openldap,
+  curl,
+  nghttp2,
+  libunwind,
+  alsa-lib,
+  libidn2,
+  rtmpdump,
+  libpsl,
+  libkrb5,
+  xkeyboard_config,
+  libsForQt5,
+  gcc,
+  pkg-config,
+  fetchFromGitHub,
+  cmake,
+  ninja,
+  wireplumber,
+  libportal,
+  xdg-desktop-portal,
+  opencv,
+  pipewire,
+  fetchgit,
+}:
+let
+  wemeet-wayland-screenshare = stdenv.mkDerivation {
+    pname = "wemeet-wayland-screenshare";
+    version = "0-unstable-2024-12-15";
+
+    src = fetchFromGitHub {
+      owner = "xuwd1";
+      repo = "wemeet-wayland-screenshare";
+      rev = "a1c8cc5a015a275256eef672f7df70a1e78e4c78";
+      hash = "sha256-uHgc4mJg0S6m9WRZ6kU1d0JwEOBDLPY1QlYCzpiEs1M=";
+      fetchSubmodules = true;
+    };
+
+    nativeBuildInputs = [
+      cmake
+      ninja
+      pkg-config
+    ];
+
+    buildInputs = [
+      wireplumber
+      libportal
+      xdg-desktop-portal
+      libsForQt5.qtwayland
+      libsForQt5.xwaylandvideobridge
+      opencv
+      pipewire
+      xorg.libXdamage
+      xorg.libXrandr
+      xorg.libX11
+    ];
+
+    dontWrapQtApps = true;
+
+    meta = {
+      description = "Hooked WeMeet that enables screenshare on Wayland";
+      homepage = "https://github.com/xuwd1/wemeet-wayland-screenshare";
+      license = lib.licenses.mit;
+      maintainers = with lib.maintainers; [ aucub ];
+    };
+  };
+  libwemeetwrap = stdenv.mkDerivation {
+    pname = "libwemeetwrap";
+    version = "1.0";
+
+    src = fetchgit {
+      url = "https://aur.archlinux.org/wemeet-bin.git";
```

Comment:
Let me see if I understand this correctly: someone in Arch created a C code that basically seems to patch some functions/symbols inside the binary? This is what I understand from a quick look here: https://aur.archlinux.org/cgit/aur.git/tree/wrap.c?h=wemeet-bin.

That's... Some dedication I think.